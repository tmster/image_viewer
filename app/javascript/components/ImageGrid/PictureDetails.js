import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function PictureDetails(props) {
  const { imageUrl, title, description, id } = props;
  return (
    <div className="row">
      <div className="col-lg-9 col-md-9 col-sm-12 col-xs-12">
        <img className="img-fluid" src={imageUrl} alt={title} />
      </div>
      <div className="col-lg-3 col-md-3 col-sm-12 col-xs-12 image-grid__details">
        <h2>{id}</h2>
        <h5>{title}</h5>
        <p>{description}</p>
        <Link to="/" className="image-grid__close" />
      </div>
    </div>
  );
}

PictureDetails.propTypes = {
  imageUrl: PropTypes.string,
  id: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string,
};

export default PictureDetails;
