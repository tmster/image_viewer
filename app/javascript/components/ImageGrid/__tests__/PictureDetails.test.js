import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter, Switch, Route } from 'react-router-dom';

import PictureDetails from '../PictureDetails';

const props = {
  id: '0',
  title: 'Test',
  description: 'Very awesome description',
  imageUrl: 'http://localhost',
};

const MockLandingPage = () => <div className="landing">Landing</div>;

it('renders all props', () => {
  const node = shallow(<PictureDetails {...props} />);
  expect(node).toMatchSnapshot();
});

it('includes link to Mission scene', () => {
  const wrapper = mount(
    <MemoryRouter>
      <PictureDetails {...props} />
      <Switch>
        <Route path="/" component={MockLandingPage} />
      </Switch>
    </MemoryRouter>
  );
  wrapper.find('a').simulate('click', { button: 0 });
  expect(wrapper.find('.landing')).toHaveLength(1);
});
