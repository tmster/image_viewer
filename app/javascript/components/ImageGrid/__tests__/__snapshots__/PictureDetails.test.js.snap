// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all props 1`] = `
<div
  className="row image-grid__overlay"
>
  <div
    className="col-lg-9 col-md-9 col-sm-12 col-xs-12 image-grid__image-overlay"
  >
    <img
      alt="Test"
      className="img-fluid image-grid__image"
      src="http://localhost"
    />
  </div>
  <div
    className="col-lg-3 col-md-3 col-sm-12 col-xs-12 image-grid__details"
  >
    <h2>
      #
      0
    </h2>
    <h5>
      Test
    </h5>
    <p>
      Very awesome description
    </p>
    <Link
      className="image-grid__close"
      to="/"
    />
  </div>
</div>
`;
